### PR TITLE
Added support for Fibocom 0x2cb7, and fixes for two issues

### PR DESCRIPTION
**Pull request by cihun**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [x] Feature
- [ ] Documentation

FM101 device-related parameters were added to quirk files of MM and fastboot plugins.

AT command receiving problem when modem opens echo in MM plug-in.

A flag is added in the MM plug-in, so that the ModemManager can uninhibit after the fastboot flash ends, so that the modem can be found again.

### CODE REVIEW

File: plugins/modem-manager/modem-manager.quirk [hunk 1]
```diff
@@ -78,3 +78,16 @@ ModemManagerFirehoseProgFile = prog_firehose_sdx6x.elf
 [PCI\VID_1EAC&PID_1007]
 Summary = Quectel RM520 firehose prog file
 ModemManagerFirehoseProgFile = prog_firehose_sdx6x.elf
+
+# Fibocom FM101
+[USB\VID_2CB7&PID_01A2]
+Summary = Fibocom Fibocom FM101-GL Module
```
Comment (hughsie) [inline plugins/modem-manager/modem-manager.quirk:84]: Fibocom Fibocom?

File: plugins/modem-manager/modem-manager.quirk [hunk 2]
```diff
@@ -78,3 +78,16 @@ ModemManagerFirehoseProgFile = prog_firehose_sdx6x.elf
...
+Flags = uninhibit-modemmanager-after-fastboot-reboot
+Flags = detach-at-fastboot-has-no-response
```
Comment (hughsie) [inline plugins/modem-manager/modem-manager.quirk:86]: does this work? I'd expected `Flags = uninhibit-modemmanager-after-fastboot-reboot,detach-at-fastboot-has-no-response`

File: plugins/modem-manager/modem-manager.quirk (hunk 1 above)
Comment (cihun) [reply plugins/modem-manager/modem-manager.quirk:84]: This is the I got using the `lsusb` query

Here are some of the results I get using `lsusb-v` :

```shell
ght@ght-Latitude-7640:~$ lsusb -v.

Bus 004 Device 030: ID 2cb7:01a2 Fibocom Fibocom FM101-GL Module
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               3.20
  bDeviceClass            0
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0         9
  idVendor           0x2cb7 Fibocom
  idProduct          0x01a2
  bcdDevice            5.04
  iManufacturer           1 Fibocom Wireless Inc.
  iProduct                2 Fibocom FM101-GL Module
  iSerial                 3 b6c08415
  bNumConfigurations      1
  Configuration Descriptor:
```

It looks like `idVendor` and `iProduct` are concatenated.

I'll modify it:

```
Summary = Fibocom FM101-GL Module
```

Thanks.

File: plugins/modem-manager/modem-manager.quirk (hunk 2 above)
Comment (cihun) [reply plugins/modem-manager/modem-manager.quirk]: It works. It has been modified in one line.

Thanks.

File: plugins/modem-manager/fu-mm-device.c [hunk 3]
```diff
@@ -625,6 +625,24 @@ fu_mm_device_at_cmd_cb(FuDevice *device, gpointer user_data, GError **error)
 	}
 	fu_dump_bytes(G_LOG_DOMAIN, "read", at_res);
 	buf = g_bytes_get_data(at_res, &bufsz);
+
+	/*
+	 * the first time the modem returns may be the command itself with one \n missing.
+	 * this is because the modem AT has enabled echo
+	 */
+	if (g_strrstr(buf, helper->cmd) != NULL && bufsz == strlen(helper->cmd) + 1) {
+		at_res = fu_io_channel_read_bytes(self->io_channel,
```
Comment (hughsie) [inline plugins/modem-manager/fu-mm-device.c:634]: don't we need a g_bytes_unref(at_res) here?

Comment (cihun) [reply plugins/modem-manager/fu-mm-device.c:635]: sorry, I was negligent.

Thanks.

**Comment by hughsie** [comment]
can you rebase against master and squash the last two commits into the existing 3 please.

**Comment by cihun** [comment]
OK, I did a rebase.

Thanks.

**Review by hughsie — APPROVED** [review]

**Pull request merged**